src/Compilers/Core/Desktop/CommandLine/CommonCommandLineParser.cs

```diff
@@ -578,10 +578,52 @@ public static IEnumerable<string> SplitCommandLineIntoArguments(string commandLi
                 }))
             .Select(arg => arg.Trim())                                                                  // Trim whitespace
             .TakeWhile(arg => (!removeHashComments || !arg.StartsWith("#", StringComparison.Ordinal)))  // If removeHashComments is true, skip all arguments after one that starts with '#'
-            .Select(arg => CondenseDoubledBackslashes(arg).Unquote())                                   // Remove quotes and handle backslashes.
+            .Select(arg => UnquoteAndUnescape(NormalizeBackslashes(arg)))                                 // Remove quotes and handle backslashes.
```

Comment:
Formatting: consider aligning the comment at the end of the line with the one on the previous line.